### PR TITLE
Add Thanos Query frontend in Query CRD and deployment

### PR DESCRIPTION
**Pull request by saswatamcode**
This adds Thanos Query frontend within query crd, which generates cache config if you don't specify

### CODE REVIEW

File: api/v1alpha1/thanosqueryfrontend_types.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+// ThanosQueryFrontendSpec defines the desired state of ThanosQueryFrontend
+type ThanosQueryFrontendSpec struct {
+	CommonThanosFields `json:",inline"`
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:default=1
+	Replicas int32 `json:"replicas,omitempty"`
+	// CompressResponses enables response compression
+	// +kubebuilder:default=true
+	CompressResponses bool `json:"compressResponses,omitempty"`
```
Comment (philipgough) [inline api/v1alpha1/thanosqueryfrontend_types.go:32]: nit: But i probably wouldnt even bother exposing this for now, we could stick with the defaults. If you want to keep it here might want to make it a pointer and optional?

Comment (saswatamcode) [reply api/v1alpha1/thanosqueryfrontend_types.go]: Let's keep it, I think I remember some issue where people were debugging with this.

**Comment by saswatamcode** [comment]
Modified this to be a nested field on Query CRD, but have kept the builder code separate as it is a separate thanos component and not a mode of another.

**Review by philipgough — APPROVED** [review]
/lgtm

**Pull request merged**